Feature/flatten demos dir

m3ller opened the pull request.

arcondello commented:
I think leaving them under `examples/`, just so that they are kept seperate from all of the ci/docs/readme/license/etc boilerplate is not a bad thing. Thoughts?

m3ller commented:
I prefer to have all the demo dirs on the top level because
(1) Once getting to the demos repo, there would never be a question of "Where are the demos?". It seems like an additional step to get to the demos repo  and then to have to discover the `examples/`
(2) Even if you know where the demos are, I feel like we'll all be doing one too many clicks to get to the file of interest.
(3) I was thinking that all the demo dirs would have their own README and doc files within their dirs; I just feel that explanations should be packaged together with their respective code. License and CI are small files; them living on the top level with the demo dirs would not bother me.

m3ller commented:
@randomir
- Renamed to `job_shop_scheduler.py`. I like `jss.py`, but I figured for a general audience, `job_shop_scheduler` is more obvious.
- I think I will hold off on naming scheme as I still have no metric for what is a more involved demo. I have removed the `demo` suffix from structural imbalance, so that things look more consistent.
